[DOC] Update Tempo and Pyroscope data source doc (#91704)
grafana/grafana: docs/sources/datasources/tempo/configure-tempo-data-source.md

## Add or modify a data source

You can use these procedures to configure a new Tempo data source or to edit an existing one.

### Add a new data source

Follow these steps to set up a new Tempo data source:

1. Select **Connections** in the main menu.
1. Enter `Tempo` in the search bar.
1. Select **Tempo**.
1. Select **Add new data source** in the top-right corner of the page.
1. On the **Settings** tab, complete the **Name**, **Connection**, and **Authentication** sections.

- Use the **Name** field to specify the name used for the data source in panels, queries, and Explore. Toggle the **Default** switch for the data source to be pre-selected for new panels.
- Under **Connection**, enter the **URL** of the Tempo instance, for example, `https://example.com:4100`.
- Complete the [**Authentication** section](#authentication).

1. Optional: Configure other sections to add capabilities to your tracing data. Refer to the additional procedures for instructions.
1. Select **Save & test**.

### Update an existing data source

To modify an existing Tempo data source:

1. Select **Connections** in the main menu.

> 1. Select Data sources to view a list of configured data sources.

Reviewer: ```suggestion
1. Select **Data sources** to view a list of configured data sources.
```

You can configure the data source using either the data source interface within Grafana or using a configuration file.
This page explains how to set up and enable the data source capabilities using Grafana.

If you are using your own installation of Grafana, you can provision the Tempo data source using a YAML configuration file.

## Before you begin

To configure a Tempo data source, you need administrator rights to your Grafana instance and a Tempo instance configured to send tracing data to Grafana.

If you are provisioning a Tempo data source, then you also need administrative rights on the server hosting your Grafana instance.
Refer to [Provision the data source](#provision-the-data-source) for next steps.

![Provisioned data source warning](/media/docs/grafana/data-sources/tempo/tempo-data-source-provisioned-error.png)

## Set up the data source

To configure basic settings for the Tempo data source, complete the following steps:

1.  Click **Connections** in the left-side menu and select **Data sources**.
1.  Enter `Tempo` in the search bar.
1.  Select **Tempo**.

> 1.  On the **Settings** tab, set the data source's basic configuration options. At a minimum, you need to complete the **Name**, **Connection**, and **Authentication** sections. The other sections provide optional capabilities.

Reviewer 2: same comment as above

Author: Done.

The Tempo data source sets how Grafana connects to your Tempo database and lets you configure features and integrations with other telemetry signals.

> You can configure the data source using either the data source interface within Grafana or using a configuration file.

Reviewer: ```suggestion
You can configure the data source using either the data source interface in Grafana or using a configuration file.
```

This page explains how to set up and enable the data source capabilities using Grafana.

> If you are using your own installation of Grafana, you can provision the Tempo data source using a YAML configuration file.

Reviewer: ```suggestion
If you're using your own installation of Grafana, you can provision the Tempo data source using a YAML configuration file.
```

## Before you begin

To configure a Tempo data source, you need administrator rights to your Grafana instance and a Tempo instance configured to send tracing data to Grafana.

> If you are provisioning a Tempo data source, then you also need administrative rights on the server hosting your Grafana instance.

Reviewer: ```suggestion
If you're provisioning a Tempo data source, then you also need administrative rights on the server hosting your Grafana instance.
```

Refer to [Provision the data source](#provision-the-data-source) for next steps.

![Provisioned data source warning](/media/docs/grafana/data-sources/tempo/tempo-data-source-provisioned-error.png)

## Set up the data source

To configure basic settings for the Tempo data source, complete the following steps:

> 1.  Click **Connections** in the left-side menu and select **Data sources**.
> 1.  Enter `Tempo` in the search bar.
> 1.  Select **Tempo**.

Reviewer: ```suggestion
1.  Click **Connections** in the main menu.
1.  Enter `Tempo` in the search bar.
1.  Select **Tempo**.
1. Click **Add new data source** in the top-right corner of the page.
```

1.  On the **Settings** tab, complete the **Name**, **Connection**, and **Authentication** sections.

> - Use the **Name** field to specify the name used for the data source in panels, queries, and Explore. Activate **Default** for the data source to be pre-selected for new panels.

Reviewer: ```suggestion
- Use the **Name** field to specify the name used for the data source in panels, queries, and Explore. Toggle the **Default** switch for the data source to be pre-selected for new panels.
```

- Complete the [**Authentication** section](#authentication).

1. Optional: Configure other sections to add capabilities to your tracing data.
1. Select **Save & test**.

This video explains how to add data sources, including Loki, Tempo, and Mimir, to Grafana and Grafana Cloud.
Tempo data source set up starts at 4:58 in the video.

{{< youtube id="cqHO0oYW6Ic" start="298" >}}

## Authentication

Use this section to select an authentication method to access the data source.

{{< admonition type="note" >}}
Use Transport Layer Security (TLS) for an additional layer of security when working with Tempo.
For additional information on setting up TLS encryption with Tempo, refer to [Configure TLS communication](https://grafana.com/docs/tempo/<TEMPO_VERSION>/configuration/network/tls/) and [Tempo configuration](https://grafana.com/docs/tempo/<TEMPO_VERSION>/configuration/).
{{< /admonition >}}

[//]: # 'Shared content for authentication section procedure in data sources'

{{< docs/shared source="grafana" lookup="datasources/datasouce-authentication.md" leveloffset="+2" version="<GRAFANA_VERSION>" >}}

> ## Streaming

Author: @adrapereira I updated this section to reflect the changes in your PR. Would you verify that these updates are correct? 

Thank you.

<!-- The traceQLStreaming toggle will be deprecated in Grafana 11.2 and removed in 11.3. -->

Streaming enables TraceQL query results to be displayed as they become available.
Without streaming, no results are displayed until all results have returned.

{{< docs/public-preview product="TraceQL streaming results" >}}

To use streaming, you need to:

- Run Tempo version 2.2 or newer, or Grafana Enterprise Traces (GET) version 2.2 or newer, or use Grafana Cloud Traces.
- For self-managed Tempo or GET instances: If your Tempo or GET instance is behind a load balancer or proxy that doesn't supporting gRPC or HTTP2, streaming may not work and should be disabled.

### Activate streaming

You can activate streaming by either setting the  `traceQLStreaming` [feature toggle](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/feature-toggles/) to true or by activating the Streaming toggle in the Tempo data source.

![Streaming section in Tempo data source](/media/docs/grafana/data-sources/tempo-data-source-streaming-v11.2.png)

If you are using Grafana Cloud, the `traceQLStreaming` feature toggle is already set to `true` by default.

> If the Tempo data source is set to allow streaming but the `traceQLStreaming` feature toggle is set to `false` in Grafana, streaming occurs.

Author: Are lines 133 and 135 correct?

| **\_\_trace.duration** | The duration of the trace.                                                                                                                                                                                                                                                                                                               |
| **\_\_trace.name**     | The name of the trace.                                                                                                                                                                                                                                                                                                                   |

## Additional settings

> Use the down-arrow to expand the **Additional settings** section to view these options.

Reviewer: Hyphen needed?
```suggestion
Use the down-arrow to expand the **Additional settings** section to view these options.
```

Author: Nope. I was thinking drop-down instead of down arrow.
